DashAI/back/dataloaders/classes/dataloader.py

```diff
@@ -0,0 +1,70 @@
+from abc import abstractmethod
+
+from datasets import ClassLabel
+
+from DashAI.back.config_object import ConfigObject
+
+
+class DataLoader(ConfigObject):
+    """
+    Abstract class with base methods for all data loaders
+    """
+
+    @abstractmethod
+    def load_data(self, dataset_path, url=None):
+        pass
+
+    @abstractmethod
+    def set_task_format(self, task):
+        pass
+
+    def split_dataset(self, dataset, params, class_column):
+        """
+        Split the dataset in train, test and validation data
+        """
+        stratify_column = None
```

Comment:
In general, there should be type and value checks on public function inputs.

```suggestion
        # input checks
        if not isinstance(params, dict):
            raise TypeError(f"params should be a string, got {type(class_column)}")
        if not isinstance(class_column, str):
            raise TypeError(
                f"stratify_column should be a string, got {type(class_column)}"
            )
        # as well the parameters in the params dict
        if not isinstance(params["stratify"], str):
            raise TypeError(
                f"params[\"stratify\"] should be a string, got {type(params['stratify'])}"
            )
        if not isinstance(params["stratify"], str):
            raise ValueError(f'params["stratify"] should be column in the dataset.')

        stratify_column = None
```

Reply:
I will add this checks, I had not considered it before because I assumed that the dictionary would come with the correct data, Which in fact may not always be the case.